Changes to patch 3.21

ufx opened the pull request:
Sorry to drop another big pull request on ya.  Mostly minor bug fixes, and a Godbert feature to aid reference navigation.

ufx committed:
SaintCoinach: Triple Triad card unlock actions.

ufx committed:
SaintCoinach: Initial patch 3.3 updates.

ufx committed:
SaintCoinach: Further patch 3.3 changes to instances and content finder.

ufx committed:
SaintCoinach: Added Item definition for the AdvancedMeldingPermitted flag.

ufx committed:
SaintCoinach: New expert delivery modifier.

ufx committed:
SaintCoinach: Fix rounding error calculating MP cost.

ufx committed:
SaintCoinach: Order sheet data definitions by index for easier comparison after updates.

ufx committed:
SaintCoinach: Patch 3.35, no changes.

ufx committed:
SaintCoinach: Add Language info for Chinese and Korean files.

ufx committed:
SaintCoinach: Fix bug comparing and losing Quad definitions on update.

ufx committed:
SaintCoinach: 2016.07.22.0000 patch, no changes.

ufx committed:
SaintCoinach: Rollback broken language updates.

ufx committed:
SaintCoinach: Added HouseRetainerPose defs.

ufx committed:
SaintCoinach: Link ActionTimeline with DefaultTalk poses.

ufx committed:
SaintCoinach: Patch 3.38, no changes.

xeropresence committed:
Add more aetheryte information

ufx committed:
Merge pull request #2 from cexikitin/master

Add more aetheryte information

ufx committed:
Godbert: Add sheet name filtering.

ufx committed:
Godbert: Handle null sheet sources.

ufx committed:
Godbert: Refrain from constructing a RawDataItemsSource when no sheet is present.

ufx committed:
Godbert: Add simple filtering to sheet data.

ufx committed:
SaintCoinach: Cache sheet name -> type match results.

ufx committed:
SaintCoinach: Use strong ISheet references to prevent redundant IO.

ufx committed:
SaintCoinach: Cache GroupDataDefinition length sum.

ufx committed:
SaintCoinach: Added GCSupplyDutyReward columns.

ufx committed:
Godbert: Fix invalid selected item crash on navigation.

ufx committed:
SaintCoinach: Added GatheringSubCategory link (folklore books) to FishParameter.

ufx committed:
SaintCoinach: Fix Quad serialization error at end of update.

ufx committed:
SaintCoinach: Patch 3.4 changes.

ufx committed:
SaintCoinach: Fix CraftAction Description reference.

ufx committed:
SaintCoinach: Addded Wondrous Tails reward definitions, and new instance icon.

ufx committed:
SaintCoinach: Use an override for ActionBase:CraftAction descriptions instead of member hiding.

ufx committed:
SaintCoinach: Added more bingo columns, and begin defining GcArmy sheets.

Kalcey committed:
Added support for a variant of .exd files containing multiple values per row+column, those will be returned as an IDictionary<int, object>.

As of now these files use this variant:
- AnimaWeapon5SpiritTalk
- Behavior
- ClassJobResident
- Credit
- DeepDungeonMap5X
- GatheringItemPoint
- GCScripShopItem
- HousingMapMarkerInfo
- MapMarker
- MobHuntOrder
- QuestClassJobReward
- Resident
- TreasureSpot
- VaseSlot
- WeatherGroup
- ZoneSharedGroup
- ZoneTimeline

Kalcey committed:
Fixed filtering.

ufx committed:
Merge pull request #3 from Kalcey/master

Support for additional .exd variant

ufx committed:
SaintCoinach: Added ClassJob and ClassBook definitions to GcArmyMemberGrow.

ufx committed:
SaintCoinach: Fix extraneous padding from the uncompressed portion of a block, resulting in sound file corruption.

ufx committed:
SaintCoinach: Added InstanceContent BGM reference.

ufx committed:
Godbert: Ensure row heights do not fluctuate when browsing data.

ufx committed:
Godbert: Remove broken expander for data views.  Use lighter grey grid lines.

ufx committed:
SaintCoinach: Added GcArmyExpeditionMemberBonus definitions.

Kalcey committed:
Fixed some issues with XivStringDecoder

ufx committed:
Merge pull request #4 from Kalcey/master

Fixed some issues with XivStringDecoder

Hezkezl committed:
Added BGM to Mounts

Adds path to the background music file when you're riding a mount.

ufx committed:
SaintCoinach: Export rawexd to the rawexd folder, so as not to conflict normal exd exports.

ufx committed:
SaintCoinach: Fix raw exd exports.

ufx committed:
SaintCoinach: 2016.10.03.0000.0000 patch, no changes.

ufx committed:
Merge pull request #6 from Hezkezl/master

Added BGM to Mounts

ufx committed:
SaintCoinach.Cmd: bgm command now extracts Orchestrion rolls.  Numbers SCDs only when >1 entries found.  Prune scd extension.

ufx committed:
SaintCoinach: Accurately scale expert delivery seal formula.

ufx committed:
SaintCoinach: Added accurate GCSupplyDutyReward definitions, thanks to Kalcey.

ufx committed:
SaintCoinach: Use GCSupplyDutyReward for expert delivery seals.

ufx committed:
SaintCoinach: Move IsAdvancedMeldingPermitted to Item to cover both meldable equipment and materia.

ufx committed:
SaintCoinach: Added EventItemHelp Description definition.

ufx committed:
SaintCoinach: Addded GcArmyExpedition definitions, thanks to ilifin.

ufx committed:
SaintCoinach: Added NPC creation definitions.

ufx committed:
SaintCoinach: Added extra EventIcon icon.

ufx committed:
DotSquish: Squish some warnings in unused code.

ufx committed:
SaintCoinach: Patch 3.41, no changes.

xeropresence committed:
Update warp and level sheet definitions

xeropresence committed:
Update MapSymbol

ufx committed:
SaintCoinach: Added additional ItemAction types.

ufx committed:
Merge pull request #7 from cexikitin/master

Update warp and level sheet definitions

ufx committed:
Merge pull request #8 from cexikitin/mapsymbol

Update MapSymbol

ufx committed:
SaintCoinach: Unify variant row and subrow column access.  Use an array for value references for faster column lookups.

ufx committed:
SaintCoinach: Fixed Icon index for MapMarker.

ufx committed:
Godbert: Format subrows as separate datagrid rows.  Join parent and subrow keys for their Key column.

ufx committed:
SaintCoinach: Added MapMarker definitions.

ufx committed:
Godbert: Fix navigation for SubRows.

ufx committed:
SaintCoinach.Cmd: Exd export formats subrows as separate CSV lines.

ufx committed:
SaintCoinach: Fix bug creating sheets with the wrong variant row type.

ufx committed:
SaintCoinach: Rolled back some memory compromises to allow the 3.45 update to proceed.

ufx committed:
SaintCoinach: Patch 3.45, no changes.

ufx committed:
SaintCoinach: Added definitions for InstanceContentBuff and a link from InstanceContent.

ufx committed:
SaintCoinach: Added accurate item bid price formula.  Rename GCTurnIn to BidFactor.

ufx committed:
SaintCoinach: Added map offsets to coordinate calculations used for some dungeons and zones.

ufx committed:
SaintCoinach: Patch 2016.11.11.0000.0000, no changes.

ufx committed:
SaintCoinach: Added GatheringItemPoint definition.

ufx committed:
SaintCoinach: Added GCScripShopItem definitions.

ufx committed:
SaintCoinach: Added Pet, PetAction definitions.

ufx committed:
SaintCoinach: Updater matches on subrow dictionaries.

ufx committed:
SaintCoinach: Do update key comparison short-circuit before collecting fields.

ufx committed:
SaintCoinach: Patch 2016.12.08.0000.0000, no changes.

ufx committed:
SaintCoinach: Added boss currency definitions to InstanceContent.

xeropresence committed:
Update ex.yaml for new information action and omen

xeropresence committed:
Update lgb entry types

ufx committed:
Merge pull request #9 from cexikitin/actionupdate

Update ex.yaml for new information on action and omen

ufx committed:
Merge pull request #10 from cexikitin/updatetypes

Update lgb entry types

ufx committed:
SaintCoinach: Patch 3.5, preliminary changes.

ufx committed:
SaintCoinach: Patch 3.5 changes so far.

ufx committed:
SaintCoinach: Fix crystal components in recipes.  Fix item bid prices.

ufx committed:
SaintCoinach: Return null row data for languages that do not have data offsets for a key.

vekien committed:
Add definition for TripleTriadCardRarity

vekien committed:
Add SecretRecipeBook and RecipeNotebookList definitions

Also links item UnlockKey to SecretRecipeBook

vekien committed:
Add RelicItem offsets

ufx committed:
SaintCoinach: Simpler route to secret recipe book unlocks.

ufx committed:
SaintCoinach: Patch 3.51, no changes.

vekien committed:
Implement "allrawexd" command

Combining Raw EXD and All EXD! Very verbous output.

ufx committed:
Godbert: Filter territories with no Bg data.

ufx committed:
Godbert: Add filtering to Territories list.

ufx committed:
SaintCoinach: Patch 3.55a, no changes.

ufx committed:
SaintCoinach: Added Satisfaction definitions.

ufx committed:
SaintCoinach: Completed SatisfactionNpc definitions.

ufx committed:
SaintCoinach: Clarify SupplyIndex and Probability<X> definitions for Satisfaction.

ufx committed:
SaintCoinach: Added DisposalShop definitions.

ufx committed:
SaintCoinach: Implement IRelationalDataRow on SubRow.

ufx committed:
SaintCoinach: Added CharaMakeCustomize defs, and ItemAction class for same.

vekien committed:
Update JobClass

Name was originally offset 24, however this is English in all 4 languages. Offset 0 has thecorrect name in all 4 languages.

ufx committed:
SaintCoinach: def consistency in ClassJob.

ufx committed:
SaintCoinach: Added StartingLevel def to ClassJob.

ufx committed:
SaintCoinach: Patch 3.56, no changes.

ufx committed:
SaintCoinach: Optimize update matching.  Levenshtein distance short-circuit, IsPrimitive ordering based on most common types, DataDefinition.Length cache.

ufx committed:
SaintCoinach: Collect compatible column indexes and comparers before evaluating definition fields to speed match detection.

ufx committed:
SaintCoinach: Updates are now parallelized.  Sort definitions before serializing for consistent order.

ufx committed:
SaintCoinach: Sort definitions by sheet name.  No other changes.

ufx committed:
SaintCoinach: Faster pre-Compile() sheet lookup in RelationDefinition, merge QuadColumnComparer into PrimitiveColumnComparer, and eliminate unnecessary conversion on primitive comparer.

ufx committed:
SaintCoinach: Fixed null reference with multirow column access.

ufx committed:
SaintCoinach: Limited retry of parallel updates that run out of memory.

ufx committed:
SaintCoinach: Added ItemSpecialBonus parameter definitions, currently used for level capped set bonuses.

ufx committed:
SaintCoinach: Added SetBonusCapped parameter type.

ufx committed:
SaintCoinach: Added index to ParameterValue base.  Required to determine set bonuses.

ufx committed:
SaintCoinach: Fixed Cmd bug when data path is specified.

ufx committed:
SaintCoinach: Patch 3.57, no changes.

ufx committed:
SaintCoinach: Added CharaMakeType definitions.

ufx committed:
SaintCoinach: Added LimitBreak defs to ClassJob.

ufx committed:
SaintCoinach: Added Element def to Action.

ufx committed:
SaintCoinach: Added TraitRecast defs.

ufx committed:
SaintCoinach.Cmd: Suppress some async warnings from Tharga commands.

ufx committed:
SaintCoinach.Cmd: Implement multiple path detection of game folder.

ufx committed:
Simplify with some C# 7 features.

ufx committed:
SaintCoinach: Added Balloon links to Behavior and ENpcBase.  Added BNpcCustomize definitions.

ufx committed:
SaintCoinach: More inline variable declarations.

ufx committed:
SaintCoinach: Added PackIdentifier for ex2.

ufx committed:
SaintCoinach: Added AquariumWater and AquariumFish definitions.

ufx committed:
SaintCoinach: Additional inline var declarations.

ufx committed:
SaintCoinach: Allow A8R8G8B8 processing on array sizes of mismatched length.

ufx committed:
SaintCoinach: Added Cost{Type} defs to Action.

ufx committed:
SaintCoinach: Added Type and SubType defs to MapMarker.  Move ToMapCoordinate to Map, and split into 2d and 3d world-space variants.  MapMarker uses the 2d variants just like FishingSpot.

ufx committed:
SaintCoinach: Fixed AST weapon type.  Added TODOs for new jobs.

ufx committed:
SaintCoinach: Added Aetheryte position definitions.  Used for calculating teleportation fees.

ufx committed:
SaintCoinach: Added CooldownGroup, BaseSpeed and LevelModifier defs.

ufx committed:
SaintCoinach: Added ClassJob param modifier defs.

ufx committed:
SaintCoinach: Rename MpCostFactor -> MpModifier as it's found to be a component of the character MP formula.  Add HpModifier def.

ufx committed:
SaintCoinach: Added Level{Start} defs to Leve.

ufx committed:
SaintCoinach: Add ArrayEventHandler link from BNpcBase.

ufx committed:
SaintCoinach: Added CustomTalk actor and map icons.

ufx committed:
SaintCoinach: Link EventIconType from Quest.

ufx committed:
SaintCoinach: Add Picture defs.

ufx committed:
SaintCoinach: Build WeatherGroups cache from WeatherGroup sheet.

ufx committed:
SaintCoinach: There are only two quest exclusions.

ufx committed:
SaintCoinach: Add equip restriction def to Item.

ufx committed:
Godbert: Make Data view the first tab.  Change MainWindow title to Godbert.

ufx committed:
SaintCoinach: Add flying defs to Mount.

ufx committed:
Godbert: Add New Window command.

vekien committed:
Add TextCommand definitions and link Emotes to it.

ufx committed:
Godbert: Implement basic data bookmarks.

ufx committed:
Godbert: Delete bookmarks with middle click, add separator, adjust format.

ufx committed:
Godbert: Hide bookmarks when there aren't any.

ufx committed:
4.0 updates, part 1.

ufx committed:
4.0 changes, part 2.

ufx committed:
4.0 updates, part 3.

ufx committed:
4.0 updates, part 4.

ufx committed:
4.0 updates, part 5.

ufx committed:
SaintCoinach: 4.0 GilShop.

ufx committed:
SaintCoinach: Changes to GilShop, tomestone reward structure.

ufx committed:
SaintCoinach: New equipment types.

ufx committed:
SaintCoinach: Collectable fixes.

ufx committed:
SaintCoinach: Several fixes to action, instances, gathering points.

TomRichter committed:
Updated MasterpieceSupplyDuty

Added Currency reference for scrips.
Corrected new Unknown column position.

ufx committed:
Link SatisfactionSupplyReward -> Currency.

ufx committed:
SaintCoinach: Fix default column name for Currency.

ufx committed:
Remove GilShop/GilShopItem.  Will have to revisit when variant2 enumerators are fixed.

ufx committed:
SaintCoinach: Adjustments to FishingSpotCategory.

ufx committed:
SaintCoinach: Remove multiref conversion until additional surrogate indexes are possible.

ufx committed:
SaintCoinach: Adjust Stain index, not sure if correct.

ufx committed:
SaintCoinach: Add BodyType appearance byte.

ufx committed:
SaintCoinach: Add BNpcAnnounceIcon icons.

ufx committed:
SaintCoinach: Use EquipmentRestriction field for model character types.  Restore action status required.

ufx committed:
SaintCoinach: Remove required status from action, this is no longer fully accurate.  Add more action cost types.  Fix EquipRestriction name.  Fix maximum value offsets.

ufx committed:
SaintCoinach: Define ActionComboRoute, and add FilterGroup to Item.

ufx committed:
SaintCoinach: Add WIP formula for calculating collectable experience.  Revise collectable level definitions.

ufx committed:
SaintCoinach: Add defs for spearfishing.

ufx committed:
SaintCoinach: Add ColorMap.  Thanks to Ioncannon for the format.

ufx committed:
SaintCoinach: Patch 2017.06.19.0000.0000 update.  Minor ordering changes.

ufx committed:
SaintCoinach: Fix RecipeNotebookList index shift.

ufx committed:
SaintCoinach: TopicSelect shops can point to GilShop too.

ufx committed:
SaintCoinach: Breaking variant2 sheet object model change.  Variant2 subrows are now enumerated directly by XivSheet2.  Parent row column access now throws an exception.

ufx committed:
SaintCoinach: Use new subrow enumerator to build weather groups.

ufx committed:
SaintCoinach: Shops always have regular items.  Add GetSheet2 for returning XivSheet2 variants without additional casting.  Reconstruct GilShop/GilShopItem.

ufx committed:
SaintCoinach: Add definitions and model for spearfishing.

ufx committed:
SaintCoinach: Add tomestone_item_conv for rewards linking to a tomestone reward index or an item.

ufx committed:
Godbert: Pass through the raw value of a column when values are null.  Useful for finding broken links.

ufx committed:
Add .editorconfig with some basic settings.

ufx committed:
Godbert: Remove unused dict check.

ufx committed:
SaintCoinach: Optimize generic reference searches by skipping lookups for 0 keys.

ufx committed:
Godbert: Keep exd pack in memory to avoid redundant IO.

ufx committed:
SaintCoinach: Rename NeedsThreeLines -> Reward{HQ} indicator.

ufx committed:
SaintCoinach: Add link to JournalSection from JournalCategory.

ufx committed:
SaintCoinach: Fix variant2 row update analysis.

ufx committed:
SaintCoinach: Patch 4.01, no changes.

ufx committed:
SaintCoinach: Treat fishing tackle as equipment, since equip level restrictions apply to them.

ufx committed:
SaintCoinach: Add unlock Item link from GatheringSubCategory.

ufx committed:
SaintCoinach: Add lots of NPC equipment offsets.  Thanks Wrakha.

ufx committed:
SaintCoinach: Add Visor definition.  Thanks Kalcey.

ufx committed:
SaintCoinach: Add IsRoleAction def.

ufx committed:
SaintCoinach: Patch 4.05 update, no changes.

ufx committed:
SaintCoinach.Cmd: Map command now supports a jpg option.

ufx committed:
SaintCoinach.Cmd: Export maps with no TerritoryType, and remove empty separator for them.

Hezkezl committed:
Update ex.yaml with new ShopItem definitions

ufx committed:
SaintCoinach: RetainerTaskParameter defs.

ufx committed:
SaintCoinach: Collectable scrip rewards also use the bonus multipliers.

ufx committed:
SaintCoinach: Fix def names for RetainerTaskParameter gathering requirements.

ufx committed:
SaintCoinach: Add def for Status MaxStacks.  Thanks to Nemekh.

Jaxelr commented:
Hello @ufx , since we see you have activity on your fork, and this current repo seems inactive, me and @rafaoneill were wondering if you were looking for help maintaining your version.

ufx commented:
Sure, feel free to send some pull requests my way!  I don't have a running list of issues but I'm generally always interested in new data offsets (from ex.yaml)

ufx committed:
SaintCoinach: Override previous definition when current definition version matches to pick up changes since history was created.

ufx committed:
SaintCoinach: Patch 4.06, no changes.

ufx committed:
SaintCoinach: Add ItemAction types for tales of adventure books and equipment coffers.

ufx committed:
SaintCoinach: GatheringitemBase links to Item and EventItem.  Link GatheringItemLevelConvertTable and add definitions for same.  Add SustainPotion and FateContentAction ItemActions.

Hezkezl committed:
Updating header for "IsHQ" for Optional Rewards

Hezkezl committed:
Adding Battleleve definitions

Hezkezl committed:
Attempt to fix problem #1

ufx committed:
SaintCoinach: Added BeastRankBonus and path to special beast tribe quest rewards.

ufx committed:
SaintCoinach: Add HQ rewards to item groups.  Thanks Hezkezl.

xeropresence committed:
Define IsPvP for actions

ufx committed:
SaintCoinach: Remove unnecessary converter on IsPvP flag.

ufx committed:
SaintCoinach: Add InterruptsCombo flag.  Thanks ilifin.

ufx committed:
SaintCoinach: Rename to PreservesCombo.

ufx committed:
SaintCoinach: Add definition serialization to JSON.

ufx committed:
SaintCoinach: Add definition deserialization from JSON.

ufx committed:
SaintCoinach: Integrate ex.yaml -> ex.json conversion.

akuseru committed:
Use reflection to get CMD commands

Add support to DataGrid to RightClick column headers to view the raw values.
 --- Image column type is currently broken, need to look into a good way to switch these to their text values without rebuilding the whole dataview.

ufx committed:
SaintCoinach: Libra fight parsing is fine.

ufx committed:
SaintCoinach: Patch 4.06a, no changes.  Order sheet definitions before serializing.

ufx committed:
SaintCoinach: Add action nature (e.g. -1 physical, 5 magical).  Rename Element to Aspect.  Add AffectsPosition index.  Thanks Maru.

ufx committed:
SaintCoinach: Rename column.

vekien committed:
Improve consistency of column names

UnlockKey > SecretRecipeBook
All "character" related levels named: ClassJobLevel, this would also match "ItemLevel" for "gear" related levels. This means those that are "Level" relate to the CSV
FishingSpot, FishParameter,  has a "Level" but unsure if this is class level? I don't do fishing
GatheringItemLevelConvertTable also has level, GatheringPointBase
MasterpieceSupplyDuty has MinLevel, is this "ClassJobLevel{Min}" ? (it also has a MaxLevel)
SpearfishingNotebook has level, this class level?

ufx committed:
SaintCoinach: Action index 39 appears to link AttackType.

vekien committed:
Rename MasterpieceSupplyDuty

vekien committed:
Update FishingSpot and GatheringPointBase to "NodeLevel"

vekien committed:
Update GatheringItemLevelConvertTable to "ItemLevel"

vekien committed:
Update SpearfishingNotebook to NodeLevel

vekien committed:
Update FishParameter to "DifficultyLevel"

vekien committed:
Suggestions to switch from "Node" to "Gathering"

Also: GatheringItemLevelConvertTable from "ItemLevel" to "GatheringItemLevel"
Also: DifficultyLevel to GatheringItemLevel

vekien committed:
Define MinionRules

ufx committed:
SaintCoinach: Add property for empty Quads.

ufx committed:
SaintCoinach: Add Quad comparison operators.

ufx committed:
SaintCoinach: Object model adjustments for recent level definition consistency changes.

ufx committed:
SaintCoinach: Add definitions for Tutorial, InstanceContentTextData, and AnimaWeaponItem.

cexikitin committed:
Show the offset for where the field is stored in the file

Hezkezl committed:
Added ExVersion def. + Quest.exh column #2 def.
